paymentsheet/src/main/java/com/stripe/android/paymentsheet/DisplayableSavedPaymentMethod.kt

```diff
@@ -50,8 +58,48 @@ internal data class DisplayableSavedPaymentMethod(
     }
 
     fun brandDisplayName(): String? {
-        val brand = paymentMethod.card?.displayBrand?.let { CardBrand.fromCode(it) }
-            ?: paymentMethod.card?.brand
-        return brand?.displayName
+        return when (savedPaymentMethod) {
+            is SavedPaymentMethod.Card -> {
+                val brand = savedPaymentMethod.card.displayBrand?.let { CardBrand.fromCode(it) }
+                    ?: savedPaymentMethod.card.brand
+                return brand.displayName
+            }
+            is SavedPaymentMethod.USBankAccount,
+            is SavedPaymentMethod.SepaDebit,
+            is SavedPaymentMethod.Unexpected -> null
+        }
     }
+
+    companion object {
+        fun create(
+            displayName: ResolvableString,
+            paymentMethod: PaymentMethod,
+            isCbcEligible: Boolean = false,
+        ): DisplayableSavedPaymentMethod {
+            val savedPaymentMethod = when (paymentMethod.type) {
+                PaymentMethod.Type.Card -> paymentMethod.card?.let { SavedPaymentMethod.Card(it) }
+                PaymentMethod.Type.USBankAccount -> paymentMethod.usBankAccount?.let {
+                    SavedPaymentMethod.USBankAccount(
+                        it
+                    )
+                }
+                PaymentMethod.Type.SepaDebit -> paymentMethod.sepaDebit?.let { SavedPaymentMethod.SepaDebit(it) }
+                else -> null
+            }
+
+            return DisplayableSavedPaymentMethod(
+                displayName = displayName,
+                paymentMethod = paymentMethod,
+                savedPaymentMethod = savedPaymentMethod ?: SavedPaymentMethod.Unexpected,
```

Comment:
If we ever use this type, we should send an unexpected error. But it'll take some work to add errorReporter everywhere that uses DisplayableSavedPaymentMethod, so I don't want to do that in this PR